ledger/src/leader_schedule_cache.rs

```diff
@@ -84,6 +86,22 @@ impl LeaderScheduleCache {
     }
 
     pub fn slot_leader_at(&self, slot: Slot, bank: Option<&Bank>) -> Option<Pubkey> {
+        let slot_epoch = self.epoch_schedule.get_epoch_and_slot_index(slot).0;
+        if slot_epoch
+            < self
+                .max_epoch
+                .read()
+                .unwrap()
+                .max(MAX_LEADER_SCHEDULE_STAKES)
+                - MAX_LEADER_SCHEDULE_STAKES
+        {
+            panic!(
```

Comment:
I'm not sure its best to crash here. 

What if someone sends our validator a Shred that's really old, it'll cause the validator to panic. 
Actually that's a bad example, I think the window_service should restructure its checks such that it doesn't risk querying a leader schedule that far back. 

Since the function returns an Option, why not just return None and log an error?

Reply:
>  I think the window_service should restructure its checks such that it doesn't risk querying a leader schedule that far back. ... Since the function returns an Option, why not just return None and log an error?

Yes, that will be what I'd write as the final form of this PR!

I'm doing `panic!` here just to know how much other code is affected by this PR's change. So I should comment as such. Sorry for being less explicit about the intention. And thanks for checking my PR!

Reply:
Gotcha! Thanks for adding the `[wip]` in the title :) Maybe a Draft PR in future experiments? 
Also the panic won't catch all possibilities. Like the RPC one I described but it's perfectly fine for what you're trying to do right now. Also imo snapshot stuff > this.